File: apk/app/src/main/java/com/example/momentra/ui/shell/group/GroupExperienceFamily.kt
package com.example.momentra.ui.shell.group

/** Group experience / purchase / living family variants — Figma populated frames. */
enum class GroupExperienceFamily {
    SHARED_GENERIC,
    WEDDING,
    HOUSE_PARTY,
    OFFICE_OUTING,
    GIFT_POOL,
    GROUP_PURCHASE,
    SHARED_ASSET,
    CUSTOM_PURCHASE,
    FLATMATES,
    FAMILY_HOUSEHOLD,
    CO_LIVING,
    CUSTOM_LIVING,
}

fun groupExperienceFamilyFor(momentTypeCode: String?): GroupExperienceFamily {
    val code = momentTypeCode?.uppercase() ?: return GroupExperienceFamily.SHARED_GENERIC
    return when {
        code.contains("WEDDING") -> GroupExperienceFamily.WEDDING
        code.contains("HOUSE_PARTY") -> GroupExperienceFamily.HOUSE_PARTY
        code.contains("OFFICE_OUTING") -> GroupExperienceFamily.OFFICE_OUTING
        code.contains("GIFT_POOL") -> GroupExperienceFamily.GIFT_POOL
        code.contains("GROUP_PURCHASE") -> GroupExperienceFamily.GROUP_PURCHASE
        code.contains("SHARED_ASSET") -> GroupExperienceFamily.SHARED_ASSET
        code.contains("COMMUNITY_PURCHASE") -> GroupExperienceFamily.CUSTOM_PURCHASE
        code.contains("FLATMATES") -> GroupExperienceFamily.FLATMATES
        code.contains("FAMILY_HOUSEHOLD") -> GroupExperienceFamily.FAMILY_HOUSEHOLD
        code.contains("CO_LIVING") -> GroupExperienceFamily.CO_LIVING
        code.contains("COMMUNITY_LIVING") -> GroupExperienceFamily.CUSTOM_LIVING
        code.contains("SHARED_LIVING") -> GroupExperienceFamily.FLATMATES
        code.contains("SHARED_RENTAL") -> GroupExperienceFamily.FLATMATES
        code == "CUSTOM" -> GroupExperienceFamily.CUSTOM_LIVING
        else -> GroupExperienceFamily.SHARED_GENERIC
    }
}

fun GroupExperienceFamily.isWedding(): Boolean = this == GroupExperienceFamily.WEDDING

fun GroupExperienceFamily.isHouseParty(): Boolean = this == GroupExperienceFamily.HOUSE_PARTY

fun GroupExperienceFamily.isOfficeOuting(): Boolean = this == GroupExperienceFamily.OFFICE_OUTING

fun GroupExperienceFamily.isThemedExperience(): Boolean =
    this == GroupExperienceFamily.HOUSE_PARTY || this == GroupExperienceFamily.OFFICE_OUTING

fun GroupExperienceFamily.isThemedPurchase(): Boolean =
    this == GroupExperienceFamily.GIFT_POOL ||
        this == GroupExperienceFamily.GROUP_PURCHASE ||
        this == GroupExperienceFamily.SHARED_ASSET ||
        this == GroupExperienceFamily.CUSTOM_PURCHASE

fun GroupExperienceFamily.isThemedLiving(): Boolean =
    this == GroupExperienceFamily.FLATMATES ||
        this == GroupExperienceFamily.FAMILY_HOUSEHOLD ||
        this == GroupExperienceFamily.CO_LIVING ||
        this == GroupExperienceFamily.CUSTOM_LIVING

fun GroupExperienceFamily.invitePeopleSubtitle(): String = when {
    isThemedLiving() -> "Share a link or add someone to this household"
    isThemedPurchase() -> "Share a link or add someone to this purchase"
    else -> "Share a link or add someone to this trip"
}

File: momentra/momentra/Shell/GroupActive/Shared/GroupExperienceFamily.swift
import SwiftUI

enum GroupExperienceFamily {
    case sharedGeneric
    case wedding
    case houseParty
    case officeOuting
    case giftPool
    case groupPurchase
    case sharedAsset
    case customPurchase
    case flatmates
    case familyHousehold
    case coLiving
    case customLiving

    static func forTypeCode(_ momentTypeCode: String?) -> GroupExperienceFamily {
        let code = (momentTypeCode ?? "").uppercased()
        if code.contains("WEDDING") { return .wedding }
        if code.contains("HOUSE_PARTY") { return .houseParty }
        if code.contains("OFFICE_OUTING") { return .officeOuting }
        if code.contains("GIFT_POOL") { return .giftPool }
        if code.contains("GROUP_PURCHASE") { return .groupPurchase }
        if code.contains("SHARED_ASSET") { return .sharedAsset }
        if code.contains("COMMUNITY_PURCHASE") { return .customPurchase }
        if code.contains("FLATMATES") { return .flatmates }
        if code.contains("FAMILY_HOUSEHOLD") { return .familyHousehold }
        if code.contains("CO_LIVING") { return .coLiving }
        if code.contains("COMMUNITY_LIVING") { return .customLiving }
        if code.contains("SHARED_LIVING") { return .flatmates }
        if code.contains("SHARED_RENTAL") { return .flatmates }
        if code == "CUSTOM" { return .customLiving }
        return .sharedGeneric
    }

    var isWedding: Bool { self == .wedding }
    var isHouseParty: Bool { self == .houseParty }
    var isOfficeOuting: Bool { self == .officeOuting }
    var isThemedExperience: Bool { self == .houseParty || self == .officeOuting }
    var isThemedPurchase: Bool {
        switch self {
        case .giftPool, .groupPurchase, .sharedAsset, .customPurchase: return true
        default: return false
        }
    }
    var isThemedLiving: Bool {
        switch self {
        case .flatmates, .familyHousehold, .coLiving, .customLiving: return true
        default: return false
        }
    }

    var invitePeopleSubtitle: String {
        if isThemedLiving { return "Share a link or add someone to this household" }
        if isThemedPurchase { return "Share a link or add someone to this purchase" }
        return "Share a link or add someone to this trip"
    }
}
